bitnami/mlflow/templates/tracking/deployment.yaml

```diff
@@ -186,10 +188,6 @@ spec:
                   key: {{ include "mlflow.v0.s3.secretAccessKeyKey" . | quote }}
             {{- end }}
             {{- end }}
-            {{- if .Values.minio.enabled }}
```

Comment:
Maybe not removing the condition (see side effects https://github.com/bitnami/charts/pull/26462#issuecomment-2146834438), but additionally checking if e.g. `externalS3.host` is defined. That would clearly indicate an external S3 is configured and the URL needs to be set. Independently from minIO being enabled.

Reply:
I see; and you are completely right. In fact, I had first fixed it in that way; and then "simplified" it; reintroducing the error.

Reply:
The new condition is equivalent to `{{- if (include "mlflow.v0.s3.enabled" .) }}`, a helper function: https://github.com/bitnami/charts/blob/1da77913298e55e7e06228deb16712dac54564fa/bitnami/mlflow/templates/_helpers.tpl#L554-L561

Maybe we can use that one instead for clarity.

---

However, I'm still missing something, the actual condition that makes `MLFLOW_S3_ENDPOINT` necessary to be set. Wouldn't this PR be a regression for https://github.com/bitnami/charts/issues/23959#issuecomment-2063750385?

Reply:
@andresbono `MLFLOW_S3_ENDPOINT` is relevant if a different S3 service from AWS is used.
This could either be minIO as provided by the chart or some other S3-comptible service. And the latter is what is currently not configurable.

As for the helper you are correct .. that could simply be used as conditional to keep the code DRY.

Reply:
ok; I changed the logic to have use the helper

Reply:
@andresbono Let's first look at what MLflow actually documents and uses to configure S3:

* https://mlflow.org/docs/latest/tracking/artifacts-stores.html#supported-storage-types-for-the-artifact-store
* https://mlflow.org/docs/latest/python_api/mlflow.environment_variables.html#mlflow.environment_variables.MLFLOW_S3_ENDPOINT_URL

Since they re using the boto3 library, it's not required to set an endpoint in case "original" AWS S3 is used, yes.
It's possible, see https://docs.aws.amazon.com/general/latest/gr/s3.html, to define regional endpoints, but it's again not required for AWS S3.

But with the current state of the PR there is a conditional asing if either `.Values.minio.enabled` **OR**  `.Values.externalS3.host` has anything but an empty string. If so an ENV var `MLFLOW_S3_ENDPOINT` is defined.

As for your question, you have me a little confused. If **neither** `minio.enabled`, not a distinct `externalS3.host` is set no ENV var `MLFLOW_S3_ENDPOINT_URL` is filled, having boto3 apply it's default, which is AWS and their S3 endpoints.

Reply:
That's right. AWS S3 does not need a region set (AWS is terrible in properly explaining this though but I double checked) -- AWS bucket names are globally unique and boto actually uses the hostname e.g. bucket hostname e.g. `bar.s3.amazonaws.com` to lookup what to connect to. There may be exceptions where an endpoint needs te set (GovCloud), but then `MLFLOW_S3_ENDPOINT` should suffice. 

See amongst others this issue: https://github.com/aws/aws-cli/issues/646